src/main/java/io/divolte/server/DivolteSchema.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2015 GoDataDriven B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server;
+
+import org.apache.avro.Schema;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+@ParametersAreNonnullByDefault
+public class DivolteSchema {
```

Comment:
Yeah, an other option is to have the configuration object be able to create factories for the required runtime stuff. Have a look at e.g. `SinkConfiguration` and `HdfsSinkConfiguration` for inspiration.